### ISSUE TITLE
Powerup suggestion: Warp Drive

### ISSUE THREAD

**Issue by DarkArc** [supertuxkart/stk-code#3888]
One thing I've noticed in SuperTuxKart is that it's possible for players to get extremely far behind and have no hope of catching up. This is fine for more competitive games, but for more casual players it poses problems.

Nintendo solves this will the bullet bill powerup, another game I played in the late 90s/early 2000s was Lego Racers, it solved this by a "Warp Turbo Boost" ([video](https://youtu.be/y55LEcUQGkE?t=7)), which effectly opened a portal, and sent you flying through it, and put you somewhere significantly further ahead of your previous position on the track.

I think something along the lines of this "Warp Turbo Boost"/"Warp Drive" may be a good way to make the game more friendly and fun for casual players, preventing them from feeling totally out of the race.

**Comment by DarkArc** [comment]
A possible mechanism for implementing this:
- Get the next player in the race, get the position they'd be placed in if requesting a reset, record it
- Display kart in warp graphic/animation for x seconds
- After animation ends place the player in the recorded position with a speed boost

This should give the player a nice catch up mechanic without actually being terribly unfair to the player in front of them.

Alternatively it could additionally pick a number `k` between `1` and `n - 1` where `n` is the current position of the kart with a heavy biased towards lower numbers, then choose the player `k` positions instead of the player immediately in front (which would be `k = 1`).

**Comment by Alayan-stk-2** [comment]
I've played Mario Kart with newbie-tiers players so I can sure see that newbies often do like getting an OP powerup like the bullet bill as they need zero skill to get some feel-good feedback. I also made the AI's rubberbanding (it can never go faster than normal, but in lower difficulties it can go slow down on purpose) more lenient because of it.

But while I do think STK ought to have an additional speed boost item for players in bad positions (the zipper speed boost need some skill to handle, newbies often either use all three zippers of a triple at once wasting its potential and/or crash because they are unable to cope with the increased speed), teleporting around the track without driving is a big no-no.

I mean, if you really want to go the extra mile and create a "super casual" race mode with hyperboosted catchup mechanisms, go for it. I'm sure I could have fun playing this sort of lottery a few times and it could be so for others.

But for the normal default mode used in story mode and in most online play ? No warp powerup will happen as long as I'm taking care of the balance.

**Comment by qwertychouskie** [comment]
I'd say that something like MK's Bullet Bill is bad because you can get almost to 1st for free.  But how about something that just gets you near the kart in 2nd-to-last without actually giving you a better position?  e.g. it would take you from last and miles behind the next kart to still last place but gives you a chance to overtake the next kart without having to catch up miles.  This is less of a problem in singleplayer due to rubberbanding, but IMO would be very useful for multiplayer.

TL;DR Bullet Bill is totally OP, but something more mild might be a great addition.

**Comment by DarkArc** [comment]
> But how about something that just gets you near the kart in 2nd-to-last without actually giving you a better position?

I think what I proposed here would give you that as it takes you to the position the player in front of you was in a few seconds ago with a speed boost to help you make up the distance if you've got the skill to control it.

> Alternatively, it could additionally pick a number k

This could exist further as a configuration option.

> I mean, if you really want to go the extra mile and create a "super casual" race mode with hyperboosted catchup mechanisms, go for it. I'm sure I could have fun playing this sort of lottery a few times and it could be so for others.

Practically something like this might have to exist with a flag/checkbox as in extreme competition you most likely wouldn't want it. Alternatively/more simply it could exist with a weight of 0 by default, and it would be up to server owners to enable it by modifying powerup.xml.

I think it would be reasonable to have as a last place position only chance item though. Even knowing what you're doing, a few bad slip ups can ruin a race in super tux kart and you're just stuck being half the track behind the other player. It's just not fun when you're all spaced out so far apart nobody can do anything to each other; it's just a rather boring loop with no action, or the occasional basketball/parachute that's too infrequent to pose a thread to your position.

**Comment by qwertychouskie** [comment]
> A possible mechanism for implementing this:
> 
> - Get the next player in the race, get the position they'd be placed in if requesting a reset, record it
> - Display kart in warp graphic/animation for x (3?) seconds
> - After animation ends place the player in the recorded position with a speed boost
> 
> This should give the player a nice catch up mechanic without actually being terribly unfair to the player in front of them.

If this item is implemented in this way, **and only given to players in last place**, I see no good reason why it couldn't be implemented.  @DarkArc If you ever want to try implementing a test version of this I'd happily help test.  A test version wouldn't need any fancy VFX, just to move the kart's position and give a speed boost.
